pype/plugins/ftrack/publish/collect_ftrack_family.py

```diff
@@ -0,0 +1,52 @@
+"""
+Requires:
+    none
+
+Provides:
+    instance     -> families ([])
+"""
+import os
+import pyblish.api
+
+from pype.lib.plugin_tools import filter_profiles
+
+
+class CollectFtrackFamily(pyblish.api.InstancePlugin):
+    """
+        Adds explicitly 'ftrack' to families to upload instance to FTrack.
+
+        Uses selection by combination of hosts/families/tasks names via
+        profiles resolution.
+
+        Triggered everywhere, checks instance against configured
+    """
+    label = "Collect Ftrack Family"
+    order = pyblish.api.CollectorOrder + 0.4999
+
+    profiles = None
+
+    def process(self, instance):
+        if self.profiles:
+            anatomy_data = instance.context.data["anatomyData"]
+            task_name = anatomy_data.get("task",
```

Comment:
You should check for task name in `instance.data["task"]` first then `context.data["task"]`  then probably stop looking for other values and expect that task is not defined. Using environment is not safe at all (especially in standalone publisher).